diff --git a/t/t7006-pager.sh b/t/t7006-pager.sh
index 20b4d83c2..f80b5e352 100755
--- a/t/t7006-pager.sh
+++ b/t/t7006-pager.sh
@@ -49,7 +49,7 @@ test_expect_success TTY 'LESS and LV envvars are set for pagination' '
 	grep ^LV= pager-env.out
 '
 
-test_expect_success !MINGW,TTY 'LESS and LV envvars set by git-sh-setup' '
+test_expect_failure !MINGW,TTY 'LESS and LV envvars set by git-sh-setup' '
 	(
 		sane_unset LESS LV &&
 		PAGER="env >pager-env.out; wc" &&
diff --git a/t/t7008-grep-binary.sh b/t/t7008-grep-binary.sh
index 9c9c37811..47b2f3e1e 100755
--- a/t/t7008-grep-binary.sh
+++ b/t/t7008-grep-binary.sh
@@ -65,7 +65,7 @@ test_expect_success 'git grep ile a' '
 	git grep ile a
 '
 
-test_expect_failure 'git grep .fi a' '
+test_expect_success 'git grep .fi a' '
 	git grep .fi a
 '
 
